<launch>
    <node pkg="nodelet" type="nodelet" name="manager" args="manager" output="screen"/>

    <param name="/use_sim_time" value="true"/>

    <include file="$(find s_map)/src/launch/rgbd2pc.launch">
        <arg name="camera_name" value="back"/>
        <arg name="manager" value="manager"/>
    </include>

    <include file="$(find s_map)/src/launch/rgbd2pc.launch">
        <arg name="camera_name" value="frontleft"/>
        <arg name="manager" value="manager"/>
    </include>

    <include file="$(find s_map)/src/launch/rgbd2pc.launch">
        <arg name="camera_name" value="frontright"/>
        <arg name="manager" value="manager"/>
    </include>

    <include file="$(find s_map)/src/launch/rgbd2pc.launch">
        <arg name="camera_name" value="left"/>
        <arg name="manager" value="manager"/>
    </include>

    <include file="$(find s_map)/src/launch/rgbd2pc.launch">
        <arg name="camera_name" value="right"/>
        <arg name="manager" value="manager"/>
    </include>

    <include file="$(find s_map)/src/launch/octomap_mapping.launch"/>    

    <include file="$(find s_map)/src/launch/process_camera.launch">
        <arg name="camera_name" value="back"/>
        <arg name="rectify_manager" value="manager"/>
        <arg name="align_manager" value="manager"/>
    </include>

    <include file="$(find s_map)/src/launch/process_camera.launch">
        <arg name="camera_name" value="frontleft"/>
        <arg name="rectify_manager" value="manager"/>
        <arg name="align_manager" value="manager"/>
    </include>

    <include file="$(find s_map)/src/launch/process_camera.launch">
        <arg name="camera_name" value="frontright"/>
        <arg name="rectify_manager" value="manager"/>
        <arg name="align_manager" value="manager"/>
    </include>

    <include file="$(find s_map)/src/launch/process_camera.launch">
        <arg name="camera_name" value="left"/>
        <arg name="rectify_manager" value="manager"/>
        <arg name="align_manager" value="manager"/>
    </include>

    <include file="$(find s_map)/src/launch/process_camera.launch">
        <arg name="camera_name" value="right"/>
        <arg name="rectify_manager" value="manager"/>
        <arg name="align_manager" value="manager"/>
    </include>

    <node pkg="s_map" type="pointcloud_merger.py" name="pointcloud_merger"  output="screen">
        <param name="world_frame" value="vision"/>
        <param name="topics" value = "/frontleft/points /frontright/points /left/points /right/points /back/points"/>
    </node>

    <node pkg="s_map" type="detection.py" name="s_map_detection_left"  output="screen">
        <param name="camera" value="left"/>
    </node>

    <node pkg="s_map" type="detection.py" name="s_map_detection_right"  output="screen">
        <param name="camera" value="right"/>
    </node>

    <node pkg="s_map" type="detection.py" name="s_map_detection_back"  output="screen">
        <param name="camera" value="back"/>
    </node>


    <node pkg="s_map" type="mapping.py" name="s_map_mapping"  output="screen"/>

</launch>